greenpeace/planet4-master-theme
# Retire ENForm block

### Summary

Removes all code exclusively required for the ENForm block. 

In addition:
- Removes relevant styles from Counter block
- Removes relevant locale text domains
- Removes twig templates that no other block uses
- Adjust locale text domains
- Added migration to clean up database from block settings and forms
---


Ref: https://jira.greenpeace.org/browse/PLANET-7673

### Testing


1. Before switching to this branch, enable _Engaging Networks_ from _Planet 4 > Features_.
2. Add some "random" keys into _Engaging Networks > Settings_. You probably already have some data there, since defaultcontent still does.
3. Browse to _Engaging Networks" > Add Form_ and create a new form. Same as above, you probably already have one as part of the defaultcontent database.
4. Switch to this branch and build assets (`npm run build:assets`).
5. Run the migration (`npx wp-env run cli wp p4-run-activator`)
6. Verify it actually removed all relevant content:
   * There should be no `feature_engaging_networks` option in that array:
   ```bash
   npx wp-env run cli wp option get planet4_features
   ```
   * You should get an `Error: Could not get 'p4en_main_settings' option. Does it exist?`
   ```bash
   npx wp-env run cli wp option get p4en_main_settings
   ```
   * You should get an empty array:
   ```bash
   npx wp-env run cli wp post list --post_type=p4en_form
   ```
7. In general, check code into detail in case something removed is still needed. I removed completely the `block_templates` and `Controllers/API` folders, since they don't seem to be used anywhere else besides ENForm block.
